src/app/state/accountHistorySlice.ts

```diff
@@ -170,15 +219,31 @@ export const accountHistorySlice = createSlice({
     resetSelectedOrdersToCancel: (state) => {
       state.selectedOrdersToCancel = {};
     },
+    setHideOtherPairs: (state, action: PayloadAction<boolean>) => {
+      state.hideOtherPairs = action.payload;
+    },
   },
 
   extraReducers: (builder) => {
-    builder.addCase(fetchAccountHistory.fulfilled, (state, action) => {
-      state.orderHistory = action.payload.data.orders;
-    });
-    builder.addCase(batchCancel.fulfilled, (state) => {
-      state.selectedOrdersToCancel = {};
-    });
+    builder
+      .addCase(fetchAccountHistory.fulfilled, (state, action) => {
+        state.orderHistory = action.payload.data.orders;
+      })
+      .addCase(batchCancel.fulfilled, (state) => {
+        state.selectedOrdersToCancel = {};
+      })
+      .addCase(
+        fetchAccountHistoryAllPairs.fulfilled,
+        (state, action: PayloadAction<SdkResult>) => {
+          if (action.payload.status === "SUCCESS") {
+            state.orderHistoryAllPairs = action.payload.data;
+          } else {
+            console.error(
+              `Failed to fetch account history: ${action.payload.message}`
+            );
+          }
```

Comment:
Inside the `fulfilled` case, only handle success case:
```suggestion
          state.orderHistoryAllPairs = action.payload.data;
```